src/constants.rs

```diff
@@ -99,3 +99,14 @@ pub const GENERATOR_H : [u8;64] = [
     0x36, 0xda, 0xc2, 0x8a, 0xf1, 0x76, 0x69, 0x68,
     0xc3, 0x0c, 0x23, 0x13, 0xf3, 0xa3, 0x89, 0x04
 ];
+
+pub const GENERATOR_PUB_J : [u8;64] = [
+    0x5f, 0x15, 0x21, 0x36, 0x93, 0x93, 0x01, 0x2a,
+    0x8d, 0x8b, 0x39, 0x7e, 0x9b, 0xf4, 0x54, 0x29,
+    0x2f, 0x5a, 0x1b, 0x3d, 0x38, 0x85, 0x16, 0xc2,
+    0xf3, 0x03, 0xfc, 0x95, 0x67, 0xf5, 0x60, 0xb8,
+    0x3a, 0xc4, 0xc5, 0xa6, 0xdc, 0xa2, 0x01, 0x59,
+    0xfc, 0x56, 0xcf, 0x74, 0x9a, 0xa6, 0xa5, 0x65,
+    0x31, 0x6a, 0xa5, 0x03, 0x74, 0x42, 0x3f, 0x42,
+    0x53, 0x8f, 0xaa, 0x2c, 0xd3, 0x09, 0x3f, 0xa4
+];
```

Comment:
No, the byte order is not correct. It should be:
```
    0xb8, 0x60, 0xf5, 0x67, 0x95, 0xfc, 0x03, 0xf3,
    0xc2, 0x16, 0x85, 0x38, 0x3d, 0x1b, 0x5a, 0x2f,
    0x29, 0x54, 0xf4, 0x9b, 0x7e, 0x39, 0x8b, 0x8d,
    0x2a, 0x01, 0x93, 0x93, 0x36, 0x21, 0x15, 0x5f,
    0xa4, 0x3f, 0x09, 0xd3, 0x2c, 0xaa, 0x8f, 0x53,
    0x42, 0x3f, 0x42, 0x74, 0x03, 0xa5, 0x6a, 0x31,
    0x65, 0xa5, 0xa6, 0x9a, 0x74, 0xcf, 0x56, 0xfc,
    0x59, 0x01, 0xa2, 0xdc, 0xa6, 0xc5, 0xc4, 0x3a,
```

Please refer to  https://github.com/garyyu/rust-secp256k1-zkp/blob/master/src/constants.rs#L130-L141 and its test: https://github.com/garyyu/rust-secp256k1-zkp/blob/master/src/demo.rs#L102-L114

Reply:
This is because this test uses `PublicKey::from_slice` which in turns uses `secp256k1_ec_pubkey_parse`. In the code in this PR I don't use the parse method, instead I set the internal bytes directly. Maybe this is not the safest method, I can change it to take for example the compressed or uncompressed serialized data, but that would mean on each call this (de)serialization has to be performed.

Reply:
To further clarify, this test passes:
```rust
let GENERATOR_PUB_J: [u8; 65] = [
	0x04,
	0xb8, 0x60, 0xf5, 0x67, 0x95, 0xfc, 0x03, 0xf3,
	0xc2, 0x16, 0x85, 0x38, 0x3d, 0x1b, 0x5a, 0x2f,
	0x29, 0x54, 0xf4, 0x9b, 0x7e, 0x39, 0x8b, 0x8d,
	0x2a, 0x01, 0x93, 0x93, 0x36, 0x21, 0x15, 0x5f,
	0xa4, 0x3f, 0x09, 0xd3, 0x2c, 0xaa, 0x8f, 0x53,
	0x42, 0x3f, 0x42, 0x74, 0x03, 0xa5, 0x6a, 0x31,
	0x65, 0xa5, 0xa6, 0x9a, 0x74, 0xcf, 0x56, 0xfc,
	0x59, 0x01, 0xa2, 0xdc, 0xa6, 0xc5, 0xc4, 0x3a,
]; // from your code
let GENERATOR_PUB_J_COMPR : [u8;33] = [
	0x02,
	0xb8, 0x60, 0xf5, 0x67, 0x95, 0xfc, 0x03, 0xf3,
	0xc2, 0x16, 0x85, 0x38, 0x3d, 0x1b, 0x5a, 0x2f,
	0x29, 0x54, 0xf4, 0x9b, 0x7e, 0x39, 0x8b, 0x8d,
	0x2a, 0x01, 0x93, 0x93, 0x36, 0x21, 0x15, 0x5f
];
let GENERATOR_PUB_J_RAW : [u8;64] = [
	0x5f, 0x15, 0x21, 0x36, 0x93, 0x93, 0x01, 0x2a,
	0x8d, 0x8b, 0x39, 0x7e, 0x9b, 0xf4, 0x54, 0x29,
	0x2f, 0x5a, 0x1b, 0x3d, 0x38, 0x85, 0x16, 0xc2,
	0xf3, 0x03, 0xfc, 0x95, 0x67, 0xf5, 0x60, 0xb8,
	0x3a, 0xc4, 0xc5, 0xa6, 0xdc, 0xa2, 0x01, 0x59,
	0xfc, 0x56, 0xcf, 0x74, 0x9a, 0xa6, 0xa5, 0x65,
	0x31, 0x6a, 0xa5, 0x03, 0x74, 0x42, 0x3f, 0x42,
	0x53, 0x8f, 0xaa, 0x2c, 0xd3, 0x09, 0x3f, 0xa4
]; // from my code

let secp = Secp256k1::with_caps(ContextFlag::Commit);
let pubkey1 = PublicKey::from_slice(&secp, &GENERATOR_PUB_J_COMPR).unwrap();
let pubkey2 = PublicKey::from_slice(&secp, &GENERATOR_PUB_J).unwrap();
let pubkey3 = PublicKey::from_secp256k1_pubkey(FFIPublicKey(GENERATOR_PUB_J_RAW));

assert_eq!(pubkey1, pubkey2);
assert_eq!(pubkey1, pubkey3);
```

Reply:
👍  
Agree to use `GENERATOR_PUB_J_RAW`, and put all above code into a test.
The suffix `RAW` is good to differentiate from the format of `G` and `H`.

Reply:
I added this test on the libsecp level, I will also add it on the rust level